packages/headless/src/api/service/insight/query/query-response.ts

```diff
@@ -0,0 +1,31 @@
+// TODO: Add interface with flesh to it
+export interface InsightQueryResponse {
+  results: SearchResult[];
+  searchUid: string;
+  totalCount: number;
+  executionReport?: Record<string, unknown>;
+  facets: Facet[];
+}
+
+interface SearchResult {
+  title: string;
+  clickUri: string;
+  excerpt: string;
+  fields: Record<string, string | number | string[]>; // TODO: make sure those types are correct
```

Comment:
Will this TODO be done separately or has this been validated already? This should be the same typing as the SearchAPI Result interface.

Reply:
I just verified again, and added the missing `boolean` type.